src/tree/updater_gpu_hist.cu

```diff
@@ -1050,6 +1050,7 @@ class GPUHistMakerSpecialised{
 
   void AllReduceHist(int nidx) {
     if (shards_.size() == 1) return;
+    dh::safe_cuda(cudaDeviceSynchronize());
```

Comment:
It may be a remnant from previous merges a number of weeks ago (making it a mistake).

@canonizer can you confirm the utility of this synchronization?

Reply:
Most likely, I added it in the process of debugging, and forgot to remove it. Feel free to do it.